Jellyfin.Server/Migrations/Routines/MigrateKeyframeData.cs

```diff
@@ -63,48 +56,33 @@ public MigrateKeyframeData(
     /// <inheritdoc />
     public void Perform()
     {
-        const int Limit = 100;
-        int itemCount = 0, offset = 0, previousCount;
+        const int Limit = 5000;
+        int itemCount = 0, offset = 0;
 
         var sw = Stopwatch.StartNew();
-        var itemsQuery = new InternalItemsQuery
-        {
-            MediaTypes = [MediaType.Video],
-            SourceTypes = [SourceType.Library],
-            IsVirtualItem = false,
-            IsFolder = false
-        };
 
         using var context = _dbProvider.CreateDbContext();
+        var records = context.BaseItems.Where(b => b.MediaType == MediaType.Video.ToString() && !b.IsVirtualItem).Count();
+        _logger.LogInformation("Checking {Count} items for importable keyframe data.", records);
+
         context.KeyframeData.ExecuteDelete();
         using var transaction = context.Database.BeginTransaction();
-        List<KeyframeData> keyframes = [];
-
         do
         {
-            var result = _libraryManager.GetItemsResult(itemsQuery);
-            _logger.LogInformation("Importing keyframes for {Count} items", result.TotalRecordCount);
-
-            var items = result.Items;
-            previousCount = items.Count;
-            offset += Limit;
+            var items = context.BaseItems.Where(b => b.MediaType == MediaType.Video.ToString() && !b.IsVirtualItem).OrderBy(e => e.Id).Skip(offset).Take(Limit).ToList();
```

Comment:
Instead of checking all BaseItems, couldn't we just read all files in the keyframe directory and then look up the respective items? This could even be batched i.e. grab X keyframe files, do `select itemid from baseitems where itemid in (insert list of ids)`, then AddRange on the keyframe data that has a corresponding baseitem

Reply:
We can't use the keyframe stuff as source because the filename is the MD4 of a combo out of file modification date and name ...

Reply:
Right. You can just select item id and path then. There's no need to fetch all columns.